sunpy/image/transform.py

```diff
@@ -118,16 +117,30 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
         # Transform the image using the skimage function
         # Image data is normalised because warp() requires an array of values
         # between -1 and 1.
-        adjusted_image = np.copy(image)
+        if np.issubdtype(image.dtype, np.integer):
+            warnings.warn("Input integer data has been cast to float64", RuntimeWarning)
+            adjusted_image = image.astype(np.float64)
+        else:
+            adjusted_image = image.copy()
+        if np.any(np.isnan(adjusted_image)) and order >= 4:
+            warnings.warn("Setting NaNs to 0 for higher-order scikit-image rotation",
+                          RuntimeWarning)
+            adjusted_image = np.nan_to_num(adjusted_image)
+
         im_min = np.nanmin(adjusted_image)
         adjusted_image -= im_min
         im_max = np.nanmax(adjusted_image)
-        adjusted_image /= im_max
-        adjusted_missing = (missing - im_min) / im_max
+        if im_max > 0:
```

Comment:
I haven't tested it, but I presume it should work fine.  The `adjusted_image -= im_min` should result in a nonnegative array (barring integer overflow, but we've already casted away from integers by this point).